src/server/scripts/Outland/zone_nagrand.cpp

```diff
@@ -636,6 +635,323 @@ class go_warmaul_prison : public GameObjectScript
         }
 };
 
+enum PlantBannerQuests
+{
+    NPC_SHADOW_COUNCIL_CREDIT_MARKER = 18388,
+    NPC_WARMAUL_OGRE_CREDIT_MARKER   = 18393,
+
+    SPELL_PLANT_WARMAUL_OGRE_BANNER  = 32307,
+    SPELL_PLANT_KIL_SORROW_BANNER    = 32314,
+
+    NPC_KIL_SORROW_SPELLBINDER       = 17146,
+    NPC_KIL_SORROW_CULTIST           = 17147,
+    NPC_KIL_SORROW_DEATHSWORN        = 17148,
+    NPC_GISELDA_THE_CRONE            = 18391,
+    NPC_WARMAUL_REAVER               = 17138,
+    NPC_WARMAUL_SHAMAN               = 18064
+};
+
+// 32307 - Plant Warmaul Ogre Banner
+class spell_q9927_plant_warmaul_ogre_banner : public SpellScript
+{
+    PrepareSpellScript(spell_q9927_plant_warmaul_ogre_banner);
+
+    void HandleHit(SpellEffIndex /*effIndex*/)
+    {
+        if (Player* caster = GetCaster()->ToPlayer())
+            caster->RewardPlayerAndGroupAtEvent(NPC_SHADOW_COUNCIL_CREDIT_MARKER, caster);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_q9927_plant_warmaul_ogre_banner::HandleHit, EFFECT_1, SPELL_EFFECT_SCRIPT_EFFECT);
+    }
+};
+
+// 32314 - Plant Kil'Sorrow Banner
+class spell_q9931_plant_kilsorrow_banner : public SpellScript
+{
+    PrepareSpellScript(spell_q9931_plant_kilsorrow_banner);
+
+    void HandleHit(SpellEffIndex /*effIndex*/)
+    {
+        if (Player* caster = GetCaster()->ToPlayer())
+            caster->RewardPlayerAndGroupAtEvent(NPC_WARMAUL_OGRE_CREDIT_MARKER, caster);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_q9931_plant_kilsorrow_banner::HandleHit, EFFECT_1, SPELL_EFFECT_SCRIPT_EFFECT);
+    }
+};
+
+
+class npc_nagrand_banner : public CreatureScript
+{
+public:
+    npc_nagrand_banner() : CreatureScript("npc_nagrand_banner") { }
+
+    struct npc_nagrand_bannerAI : public ScriptedAI
+    {
+        npc_nagrand_bannerAI(Creature* creature) : ScriptedAI(creature), bannered(false)
+        {
+            scheduler.SetValidator([this]
+            {
+                return !me->HasUnitState(UNIT_STATE_CASTING);
+            });
+        }
+
+        void Reset() override
+        {
+            scheduler.CancelAll();
+        }
+
+        void SpellHit(Unit* /*caster*/, SpellInfo const* spellInfo) override
+        {
+            if (spellInfo->Id == SPELL_PLANT_WARMAUL_OGRE_BANNER || spellInfo->Id == SPELL_PLANT_KIL_SORROW_BANNER)
+                bannered = true;
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            if (!UpdateVictim())
+                return;
+
+            scheduler.Update(diff, [this]
+            {
+                DoMeleeAttackIfReady();
+            });
+        }
+
+        bool bannered;
```

Comment:
please expose this field with an accessor method like bool IsBannered() and make the field private